sdks/go/pkg/beam/io/spannerio/read_batch_integration_test.go

```diff
@@ -0,0 +1,126 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spannerio
+
+import (
+	"cloud.google.com/go/spanner"
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	"context"
+	"flag"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/testing/passert"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/testing/ptest"
+	adminpb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+	"reflect"
+	"testing"
+)
+
+var (
+	integrationTests = flag.Bool("int", false, "")
+)
+
+func TestExampleQueryBatch(t *testing.T) {
+	if !*integrationTests {
+		t.Skip("Not running in integration test mode.")
+	}
```

Comment:
Please put integration tests in an appropriate directory in https://github.com/apache/beam/tree/master/sdks/go/test/integration/io and use the existing infrastructure for integration tests. This will allow the test to run on several runners.

Although, I'll note that as written, this test won't work as it's using a fake client/server, like the other tests, but notably, the [spannertest fake doesn't support partitions at this time](https://pkg.go.dev/cloud.google.com/go/spanner/spannertest#section-readme)

It would be important for any integration tests against real spanner would need to take care to clean up after itself so we don't waste beam project resources.

Reply:
I've moved this to the integration test directory and corrected the issue of using the fake client/server.